src/miner.cpp

```diff
@@ -374,7 +381,7 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn)
 
         // Fill in header
         pblock->hashPrevBlock  = pindexPrev->GetBlockHash();
-        pblock->hashReserved   = uint256();
+        pblock->hashFinalSaplingRoot   = sapling_tree.root();
```

Comment:
This is the cause of the test failure; the test calls this function and then sticks in the nonce and solution previously calculated, in order to create a valid block.

I'm not against making this change unconditional, since `hashReserved` is explicitly an unconstrained field before Sapling, and any mining software that assumes otherwise is broken. (A while back I asked several mining software developers about this, and they all IIRC confirmed they correctly read `hashReserved` from the Stratum protocol. The open-source mining pool software Z-NOMP, OTOH, I think I've seen ignore this output and create blocks with an all-zero field, but as a result this change won't affect it, and it will need to be updated either way before Sapling activates.)

The simplest solution is likely to just set `pblock->hashFinalSaplingRoot` back to `uint256()` after calling this function, in order to make the test happy.